### PR TITLE
Added definition lists

### PR DESCRIPTION
**Pull request by ahalbert**
#15 I put the =defn statements in a

```
<dl>
<dt>Word</dt>
<dd>defnition</dd> 
</dl>
```

table.

### CODE REVIEW

File: lib/Pod/To/HTML.pm [hunk 1]
```diff
@@ -45,8 +45,9 @@ my @body;
 my @footnotes;
 my %crossrefs;
 
- sub Debug(Callable $)  { }         # Disable debug code
-#sub Debug(Callable $c) { $c() }    # Enable debug code
+#sub Debug(Callable $)  { }         # Disable debug code
+sub Debug(Callable $c) { $c() }    # Enable debug code
```
Comment (zoffixznet) [inline lib/Pod/To/HTML.pm]: I'm unfamiliar with this codebase, but this looks like the Debug mode is still enabled?

Comment (ahalbert) [inline lib/Pod/To/HTML.pm]: Ooh yeah, I need to go back and look at the diff again.

**Comment by zoffixznet** [comment]
Thanks for fixing that. I ran some tests and something doesn't seem to be working. Here's the POD file I used:

```
=begin pod

=defn  MAD
Affected with a high degree of intellectual independence.

=defn  MEEKNESS
Uncommon patience in planning a revenge that is worth while.

=defn
MORAL
Conforming to a local and mutable standard of right.
Having the quality of general expediency.

=end pod
```

With your PR, the page ended up with 3 empty definition lists:

```html
<div class="pod-body
 no-toc
">
<dl><dt></dt><dd></dd></dl>
<dl><dt></dt><dd></dd></dl>
<dl><dt></dt><dd></dd></dl>

</div>
```

I then went in and dumped the `$node` inside the `when 'defn'` block and this is what it gave me:

```perl6
Pod::Block::Named defn = Pod::Block::Named.new(name => "defn", config => {}, contents => [Pod::Block::Para.new(config => {}, contents => ["MAD Affected with a high degree of intellectual independence."])])
Pod::Block::Named defn = Pod::Block::Named.new(name => "defn", config => {}, contents => [Pod::Block::Para.new(config => {}, contents => ["MEEKNESS Uncommon patience in planning a revenge that is worth while."])])
Pod::Block::Named defn = Pod::Block::Named.new(name => "defn", config => {}, contents => [Pod::Block::Para.new(config => {}, contents => ["MORAL Conforming to a local and mutable standard of right. Having the quality of general expediency."])])
```

So it seems like Rakudo doesn't parse definition lists correctly yet. As the terms are not separated from the definitions. This is Rakudo version 2016.10-15-g43dbc96 built on MoarVM version 2016.10-9-g0402247

**Comment by zoffixznet** [comment]
@ahalbert ping. Any update on the issue above?

**Comment by tbrowder** [comment]
Bug is rakudo RT #126651.

**Comment by JJ** [comment]
The bug seems to be still open https://rt.perl.org/Ticket/Display.html?id=126651

**Comment by tbrowder** [comment]
I am working on the proper handling of pod =defn by rakudo.

**Comment by JJ** [comment]
Since definition lists are about to be included in Rakudo, can you please check this for conflicts and resubmit it?

**Comment by jonathanstowe** [comment]
I've fixed the conflict, but would like to add at least one test before I push it.

**Comment by jonathanstowe** [comment]
Actually that's not right,  just trying to work out now.

**Comment by jonathanstowe** [comment]
Oh, this expects a ```Pod::Block::Named``` but @tbrowder's change made it a  ```Pod::Defn```  so it won't work. But I'm here now so it would be a shame not to have this work :-)

**Comment by tbrowder** [comment]
I changed the class to Pod::Defn to be in sync with the other list classes Pod::Item and Pod::Heading. They are special cases of named blocks.

**Comment by jonathanstowe** [comment]
No it's fine, just moved the code from the ````when 'defn'``` into its own ````node2html```` and it works :)

I'll push in a bit.

The only thing as it stands is each ```=defn``` will be in its own ```<dl>```` for the time being (as the would in the original PR,) so someone might want to take a view on whether it would be desirable to merge consecutive ones into a single list.`

**Comment by JJ** [comment]
@jonathanstowe thanks!

**Comment by tbrowder** [comment]
It seemes to me that, without other hints from the author, consecutive =defns should be one list.

**Comment by JJ** [comment]
It's not going to be a big difference, except if some CSS is used. We can
post-process it, it's not a big deal. Maybe reuse code from =item

**Comment by jonathanstowe** [comment]
I've merged as is, but yes altering ````assemble-list-items```` to deal with the Defn as well as the items would probably be the way to go.

**Comment by JJ** [comment]
@tbrowder you can probably do that using CSS. As long as markup is OK, any
other content or styling can be added afterwards.

**Comment by jonathanstowe** [comment]
I've made s a small change to have the Defns merged. It may need more tests and I'll have a look later.I

**Comment by JJ** [comment]
Thanks, Jonathan.

**Pull request merged**